v1/build_config.go

```diff
@@ -126,7 +126,7 @@ func (c *Client) CreateBuildConfig(user, name string) (*BuildConfig, error) {
 //
 // Actual API: "Create Build Config Version"
 func (c *Client) UploadBuildConfigVersion(v *BuildConfigVersion, metadata map[string]interface{},
-	data io.Reader, size int64) error {
+	vars map[string]string, data io.Reader, size int64) error {
```

Comment:
Could we borrow a book from the aws-sdk-go here and make this a struct input instead? That will give us the flexibility to add new fields in the future without breaking the signature.